File: test_resources/kotlin/struct_with_positions.kt
data class Test(
    val touchY: Float,
    val touchX: Float,
    val touchStatus: TouchStatus,
) {
    companion object {
        fun createDefault(): Test = Test(
            touchY = 0f,
            touchX = 0f,
            touchStatus = TouchStatus.createDefault(),
        )

        fun readFromBuffers(reader: Long): Test {
            val touchY = Float.readFromBuffers(reader)
            val touchX = Float.readFromBuffers(reader)
            val touchStatus = TouchStatus.readFromBuffers(reader)

            return Test(
                touchY = touchY,
                touchX = touchX,
                touchStatus = touchStatus,
            )
        }

        fun skipInBuffers(reader: Long, count: Int) {
            for (i in 0..<count) {
                Float.readFromBuffers(reader)
                Float.readFromBuffers(reader)
                TouchStatus.readFromBuffers(reader)
            }
        }
    }

    fun writeToBuffers(writer: Long) {
        touchY.writeToBuffers(writer)
        touchX.writeToBuffers(writer)
        touchStatus.writeToBuffers(writer)
    }
}

File: test_resources/dart/struct_with_positions.dart
class Test {
  const Test({
    required this.touchY,
    required this.touchX,
    required this.touchStatus,
  });

  const Test.createDefault()
      : touchY = 0.0,
        touchX = 0.0,
        touchStatus = const TouchStatus.createDefault();

  final double touchY;
  final double touchX;
  final TouchStatus touchStatus;
}

class TestBuffersFactory implements BuffersFactory<Test> {
  const TestBuffersFactory();

  @override
  Test createDefault() => const Test.createDefault();
}
